scripts/loc.sh

```
@@ -5,41 +5,179 @@
+    if [[ -z "$SOURCE_DIR" ]]; then
+        SOURCE_DIR="."
+    fi
+fi
+
+echo "📊 Lines of Code Breakdown ($SOURCE_DIR directory)"
+echo "========================================================================"
+
+# Function to count lines in files
+count_lines() {
+    local pattern="$1"
+    local files=$(find "$SOURCE_DIR" -type f -name "$pattern" ! -path "*/__pycache__/*" ! -path "*/.pytest_cache/*" ! -path "*/node_modules/*" 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)